quarkus-bootstrap-maven-plugin:build-tree requires installed project in local maven repository

quarkus-bootstrap-maven-plugin:build-tree requires installed project in local maven repository, otherwise it fails with message similar to this: Failed to resolve application model org.acme:getting-started::jar:1.0-SNAPSHOT dependencies: Failed to resolve artifact org.acme:getting-started:jar:1.0-SNAPSHOT: Could not find artifact org.acme:getting-started:jar:1.0-SNAPSHOT

Quarkus 1.3.4 based quarkus-bootstrap-maven-plugin:build-tree command doesn't require to install local project into local maven repository first.

Workaround is to call `mvn clean install` before calling quarkus-bootstrap-maven-plugin:build-tree.
It's a regression against Quarkus 1.3

Reproducer:
```
 mvn io.quarkus:quarkus-maven-plugin:1.8.1.Final:create \
 -DprojectGroupId=org.acme \
 -DprojectArtifactId=getting-started \
 -DclassName="org.acme.getting.started.GreetingResource" \
 -Dpath="/hello"
 cd getting-started
```
```
OK - mvn io.quarkus:quarkus-bootstrap-maven-plugin:1.3.4.Final:build-tree -Dmaven.repo.local=/Users/rsvoboda/Downloads/maven-repository-1.3/ -Dquarkus.platform.version=1.3.4.Final -Dquarkus-plugin.version=1.3.4.Final
```
```
FAIL - mvn io.quarkus:quarkus-bootstrap-maven-plugin:1.7.3.Final:build-tree -Dmaven.repo.local=/Users/rsvoboda/Downloads/maven-repository-1.7/ -Dquarkus.platform.version=1.7.3.Final -Dquarkus-plugin.version=1.7.3.Final

...

[ERROR] Failed to execute goal io.quarkus:quarkus-bootstrap-maven-plugin:1.7.3.Final:build-tree (default-cli) on project
  getting-started: Failed to resolve application model org.acme:getting-started::jar:1.0-SNAPSHOT dependencies:
  Failed to resolve artifact org.acme:getting-started:jar:1.0-SNAPSHOT: 
  Could not find artifact org.acme:getting-started:jar:1.0-SNAPSHOT -> [Help 1]
```

Yes I did, it didn't work too.

Here is the output:
```
mvn io.quarkus:quarkus-bootstrap-maven-plugin:1.8.1.Final:build-tree -Dmaven.repo.local=/Users/rsvoboda/Downloads/maven-repository-1.8/
[INFO] Scanning for projects...
[INFO]
[INFO] ----------------------< org.acme:getting-started >----------------------
[INFO] Building getting-started 1.0-SNAPSHOT
[INFO] --------------------------------[ jar ]---------------------------------
[INFO]
[INFO] --- quarkus-bootstrap-maven-plugin:1.8.1.Final:build-tree (default-cli) @ getting-started ---
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  0.892 s
[INFO] Finished at: 2020-09-23T08:49:45+02:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal io.quarkus:quarkus-bootstrap-maven-plugin:1.8.1.Final:build-tree (default-cli) on project getting-started: Failed to resolve application model org.acme:getting-started::jar:1.0-SNAPSHOT dependencies: Failed to resolve artifact org.acme:getting-started:jar:1.0-SNAPSHOT: Could not find artifact org.acme:getting-started:jar:1.0-SNAPSHOT -> [Help 1]
[ERROR]
```

--- a/independent-projects/bootstrap/maven-plugin/src/main/java/io/quarkus/maven/AbstractTreeMojo.java
+++ b/independent-projects/bootstrap/maven-plugin/src/main/java/io/quarkus/maven/AbstractTreeMojo.java
@@ -20,7 +20,8 @@ public class AbstractTreeMojo extends AbstractMojo {
 
     @Override
     public void execute() throws MojoExecutionException, MojoFailureException {
-        final AppArtifact appArtifact = new AppArtifact(project.getGroupId(), project.getArtifactId(), project.getVersion());
+        final AppArtifact appArtifact = new AppArtifact(project.getGroupId(), project.getArtifactId(), null, "pom",
+                project.getVersion());
         final BootstrapAppModelResolver modelResolver;
         try {
             modelResolver = new BootstrapAppModelResolver(resolver());

--- a/independent-projects/bootstrap/maven-plugin/src/test/resources/test-app-1.jar.build-tree
+++ b/independent-projects/bootstrap/maven-plugin/src/test/resources/test-app-1.jar.build-tree
@@ -1,4 +1,4 @@
-[info] io.quarkus.bootstrap.test:test-app:jar:1
+[info] io.quarkus.bootstrap.test:test-app:pom:1
 [info] ├─ io.quarkus.bootstrap.test:artifact-with-classifier:jar:classifier:1 (compile)
 [info] ├─ io.quarkus.bootstrap.test:test-ext2-deployment:jar:1 (compile)
 [info] │  ├─ io.quarkus.bootstrap.test:test-ext2:jar:1 (compile)

--- a/independent-projects/bootstrap/maven-plugin/src/test/resources/test-app-1.jar.dev-mode-tree
+++ b/independent-projects/bootstrap/maven-plugin/src/test/resources/test-app-1.jar.dev-mode-tree
@@ -1,4 +1,4 @@
-[info] io.quarkus.bootstrap.test:test-app:jar:1
+[info] io.quarkus.bootstrap.test:test-app:pom:1
 [info] ├─ io.quarkus.bootstrap.test:artifact-with-classifier:jar:classifier:1 (compile)
 [info] ├─ io.quarkus.bootstrap.test:test-ext2-deployment:jar:1 (compile)
 [info] │  ├─ io.quarkus.bootstrap.test:test-ext2:jar:1 (compile)

